source/conservation/ConservedMoietyConverter.cpp

```diff
@@ -812,10 +813,8 @@ static void conservedMoietyCheck(const SBMLDocument *doc)
             if (!ass->isSetMath()) {
                 continue;
             }
-            const SBase *element = const_cast<Model*>(model)->getElementBySId(
-                    ass->getVariable());
 
-            const Species *species = dynamic_cast<const Species*>(element);
+            const Species* species = model->getSpecies(ass->getVariable());
```

Comment:
I think we can improve this further by storing the libsbml::Species in maps in LLVMDataModelSymbols. Currently working on this now, looks to be working so will push to this branch once ready.